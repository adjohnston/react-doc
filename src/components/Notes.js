import React from 'react'
import PropTypes from 'prop-types'

const FallbackTemplate = (notes) => (!!notes && (
  <section>
    <h2>
      Notes
    </h2>

    {Object.keys(notes).map(group => (
      <div
        key={group}>
        <h3>
          {group}
        </h3>
        {notes[group]}
      </div>
    ))}
  </section>
)) || null

const Notes = ({ children, notes }) => (
  (children && notes)
    ? children(notes)
    : FallbackTemplate(notes)
)

Notes.propTypes = {
  notes: PropTypes.object
}

export default Notes
