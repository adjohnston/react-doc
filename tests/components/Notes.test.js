import React from 'react'
import renderer from 'react-test-renderer'
import Notes from 'src/components/Notes'

test(`it doesn't render without notes`, () => {
  const tree = renderer
    .create(<Notes />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it renders correctly', () => {
  const props = {
    notes: {}
  }

  const tree = renderer
    .create(<Notes {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly when given a template', () => {
  const mockProps = {
    notes: {
      'Developer Notes': 'I am a developer',
      'Designer Notes': 'I am a designer'
    },
    children: (notes) => (notes && notes.map(note => {
      return (
        <div>
          {note}
        </div>
      )
    }))
  }

  const tree = renderer
    .create(<Notes {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it renders groups of notes using strings', () => {
  const props = {
    notes: {
      'Developer Notes': 'I am a developer',
      'Designer Notes': 'I am a designer'
    }
  }

  const tree = renderer
    .create(<Notes {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it renders groups of notes using components', () => {
  const props = {
    notes: {
      'Developer Notes': (
        <div>I am a developer</div>
      ),
      'Designer Notes': (
        <div>I am a designer</div>
      )
    }
  }

  const tree = renderer
    .create(<Notes {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
