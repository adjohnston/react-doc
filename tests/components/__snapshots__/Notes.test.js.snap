// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it doesn't render without notes 1`] = `null`;

exports[`it renders correctly 1`] = `
<section>
  <h2>
    Notes
  </h2>
</section>
`;

exports[`it renders groups of notes using components 1`] = `
<section>
  <h2>
    Notes
  </h2>
  <div>
    <h3>
      Developer Notes
    </h3>
    <div>
      I am a developer
    </div>
  </div>
  <div>
    <h3>
      Designer Notes
    </h3>
    <div>
      I am a designer
    </div>
  </div>
</section>
`;

exports[`it renders groups of notes using strings 1`] = `
<section>
  <h2>
    Notes
  </h2>
  <div>
    <h3>
      Developer Notes
    </h3>
    I am a developer
  </div>
  <div>
    <h3>
      Designer Notes
    </h3>
    I am a designer
  </div>
</section>
`;

exports[`it should render correctly when given a template 1`] = `
<div>
  <h2>
    Notes
  </h2>
  <div>
    Developer Notes
  </div>
  <div>
    Designer Notes
  </div>
</div>
`;
